config/config_test.go

```diff
@@ -0,0 +1,211 @@
+			wanted := config.Config{Namespace: "some namespace"}
+			cfg := wanted
+			config.FromEnv(&cfg)
+			So(cfg, ShouldEqual, wanted)
+		})
+
+		Convey("Setting HUMAN_LOG updates pretty config", func() {
+			cfg := config.Config{}
+
+			tests := []struct {
+				value  string
+				wanted bool
+			}{
+				{"1", true},
+				{"0", false},
+				{"t", true},
+				{"f", false},
+				{"true", true},
+				{"false", false},
+			}
+
+			for _, test := range tests {
+				Convey("When HUMAN_LOG="+test.value, func() {
+					os.Setenv("HUMAN_LOG", test.value)
+					config.FromEnv(&cfg)
+					So(cfg.Pretty, ShouldEqual, test.wanted)
+				})
+			}
+		})
+
+		Convey("Setting LOG_LEVEL updates level in config", func() {
+			initialLevel := slog.Level(327)
+			cfg := config.Config{Level: initialLevel}
+
+			tests := []struct {
+				value  string
+				wanted slog.Level
+			}{
+				{"DEBUG", log.LevelDebug},
+				{"debug", log.LevelDebug},
+				{"Debug", log.LevelDebug},
+				{"-4", log.LevelDebug},
+
+				{"INFO", log.LevelInfo},
+				{"info", log.LevelInfo},
+				{"Info", log.LevelInfo},
+				{"0", log.LevelInfo},
+
+				{"WARN", log.LevelWarn},
+				{"warn", log.LevelWarn},
+				{"Warn", log.LevelWarn},
+				{"4", log.LevelWarn},
+
+				{"ERROR", log.LevelError},
+				{"error", log.LevelError},
+				{"Error", log.LevelError},
+				{"8", log.LevelError},
+
+				{"FATAL", log.LevelFatal},
+				{"fatal", log.LevelFatal},
+				{"Fatal", log.LevelFatal},
+				{"12", log.LevelFatal},
+
+				{"99", slog.Level(99)},
+
+				// Invalid config leaves level in config unchanged
+				{"nonsense", initialLevel},
+			}
+
+			for _, test := range tests {
+				Convey("When HUMAN_LOG="+test.value, func() {
+					cfg.Level = initialLevel
+					os.Setenv("LOG_LEVEL", test.value)
+					config.FromEnv(&cfg)
+					So(cfg.Level, ShouldEqual, test.wanted)
+				})
+			}
+		})
+
+	})
+}
+
+func TestFromOptions(t *testing.T) {
+	const (
+		ns = "some_namespace"
+	)
+
+	Convey("with some predefined config options", t, func() {
+		prettyTrueOpt := func(cfg *config.Config) { cfg.Pretty = true }
+		prettyFalseOpt := func(cfg *config.Config) { cfg.Pretty = false }
+		nsOpt := func(cfg *config.Config) { cfg.Namespace = ns }
+
+		Convey("no opts leaves config unchanged", func() {
+			cfg := config.FromOptions()
+			So(cfg, ShouldNotBeNil)
+			So(*cfg, ShouldEqual, config.Config{})
+		})
+
+		Convey("single opt should set appropriate value", func() {
+			cfg := config.FromOptions(prettyTrueOpt)
+			So(cfg, ShouldNotBeNil)
+			So(cfg.Pretty, ShouldBeTrue)
+		})
+
+		Convey("duplicate opts should use last value", func() {
+			cfg := config.FromOptions(prettyTrueOpt, prettyFalseOpt)
+			So(cfg, ShouldNotBeNil)
+			So(cfg.Pretty, ShouldBeFalse)
+		})
+
+		Convey("multiple opts should not conflict", func() {
+			cfg := config.FromOptions(prettyTrueOpt, nsOpt)
+			So(cfg, ShouldNotBeNil)
+			So(cfg.Pretty, ShouldBeTrue)
+			So(cfg.Namespace, ShouldEqual, ns)
+		})
+	})
```

Comment:
Think we should be consistent on our text casing throughout convey statements, I'm not that fussed between the two.

Reply:
Capitalised all lower case starting Convey lines (or amended where necessary).